app/src/main/java/org/piramalswasthya/cho/network/TokenRefreshProviderImpl.kt

```
@@ -0,0 +1,77 @@
+                preferenceDao.registerJWTAmritToken(data.getString("jwtToken"))
+                val token = data.getString("key")
+                TokenInsertTmcInterceptor.setToken(token)
+                preferenceDao.registerPrimaryApiToken(token)
+                Timber.d("TokenRefreshProvider: Token refreshed for user $userName")
+                true
+            } else {
+                val errorMessage = responseBody.optString("errorMessage", "Unknown error")
+                Timber.w("TokenRefreshProvider: API error $responseStatusCode - $errorMessage")
+                false
+            }
+        } catch (e: SocketTimeoutException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)